src/index.test.mjs

```diff
@@ -65,64 +65,171 @@ test("astroRehypeRelativeMarkdownLinks", async (t) => {
     },
   );
 
-  await t.test("should keep query and fragment", async () => {
-    const input = '<a href="./fixtures/test.md?q=q#hash">foo</a>';
+  await t.test(
+    "should keep query and fragment for file paths that exist",
+    async () => {
+      const input = '<a href="./fixtures/test.md?q=q#hash">foo</a>';
+      const { value: actual } = await rehype()
+        .use(testSetupRehype)
+        .use(astroRehypeRelativeMarkdownLinks, { contentPath: "src" })
+        .process(input);
+
+      const expected =
+        '<html><head></head><body><a href="/fixtures/test?q=q#hash">foo</a></body></html>';
+
+      assert.equal(actual, expected);
+  });
+
+  await t.test(
+    "should prefix base to output on file paths that exist",
+    async () => {
+      const input = '<a href="./fixtures/test.md">foo</a>';
+      const { value: actual } = await rehype()
+        .use(testSetupRehype)
+        .use(astroRehypeRelativeMarkdownLinks, {
+          contentPath: "src",
+          basePath: "/testBase",
+        })
+        .process(input);
+
+      const expected =
+        '<html><head></head><body><a href="/testBase/fixtures/test">foo</a></body></html>';
+
+      assert.equal(actual, expected);
+  });
+
+  await t.test(
+    "should not replace path if relative file does not exist",
+    async () => {
+      const input = '<a href="./fixtures/does-not-exist.md">foo</a>';
+      const { value: actual } = await rehype()
+        .use(testSetupRehype)
+        .use(astroRehypeRelativeMarkdownLinks, { contentPath: "src" })
+        .process(input);
+
+      const expected =
+        '<html><head></head><body><a href="./fixtures/does-not-exist.md">foo</a></body></html>';
+
+      assert.equal(actual, expected);
+    },
+  );
+
+  await t.test(
+    "should transform file paths that exist with non alphanumeric characters",
+    async () => {
+      const input = '<a href="./fixtures/test%20(non-alpha).md">foo</a>';
+      const { value: actual } = await rehype()
+        .use(testSetupRehype)
+        .use(astroRehypeRelativeMarkdownLinks, { contentPath: "src" })
+        .process(input);
+
+      const expected =
+        '<html><head></head><body><a href="/fixtures/test-non-alpha">foo</a></body></html>';
+
+      assert.equal(actual, expected);
+    },
+  );
+
+  await t.test("should not replace absolute path if file exists", async () => {
+    const absolutePath = path.resolve("./fixtures/test.md");
+    const input = `<a href="${absolutePath}">foo</a>`;
+    const { value: actual } = await rehype()
+      .use(testSetupRehype)
+      .use(astroRehypeRelativeMarkdownLinks, { contentPath: "src" })
+      .process(input);
+
+    const expected = `<html><head></head><body><a href="${absolutePath}">foo</a></body></html>`;
+
+    assert.equal(actual, expected);
+  });
```

Comment:
Can you please help me understand the intent behind this test? It's not a use case I've thought of.

From what I can understand, it takes an absolute path and outputs an absolute path.

Reply:
Correct.  There is logic in the code to validate that its a relative file that exists (`isValidRelativeFile`) but there weren't any tests previously that ensured we didn't transform non-relative paths.